tests/Requests/RequestTest.php

```diff
@@ -5,61 +5,39 @@
 namespace OpenEuropa\EPoetry\Tests\Requests;
 
 use GuzzleHttp\Psr7\Response;
-use OpenEuropa\EPoetry\Type\ContactPersonIn;
-use OpenEuropa\EPoetry\Type\Contacts;
+use OpenEuropa\EPoetry\Serializer\RequestsSerializer;
 use OpenEuropa\EPoetry\Type\CreateRequests;
 use OpenEuropa\EPoetry\Type\CreateRequestsResponse;
-use OpenEuropa\EPoetry\Type\LinguisticRequestIn;
-use OpenEuropa\EPoetry\Type\RequestGeneralInfoIn;
 
 /**
  * @internal
  * @coversNothing
  */
-final class RequestTest extends AbstractTest
+final class RequestTest extends AbstractRequestTest
 {
+    /**
+     * Data provider.
+     *
+     * @return mixed
+     */
+    public function requestSendingCases()
+    {
+        return $this->getFixture('request-test.yml');
+    }
+
     /**
      * Test a SOAP request.
+     *
+     * @dataProvider requestSendingCases
+     *
+     * @param mixed $input
      */
-    public function testRequestSending()
+    public function testRequestSending($input)
     {
-        // Generate General Info.
-        $generalInfo = new RequestGeneralInfoIn();
-
-        $generalInfo->setTitle('Test')
-            ->setInternalReference('1')
-            ->setInternalTechnicalId('1')
-            ->setRequestedDeadline(new \DateTime('2020-02-02'))
-            ->setSensitive(false)
-            ->setDocumentToBeAdopted(true)
-            ->setDecideReference('decideReference')
-            ->setSentViaRUE(true)
-            ->setDestinationCode('PUBLIC')
-            ->setSlaAnnex('')
-            ->setSlaCommitment('')
-            ->setComment('')
-            ->setOnBehalfOf('')
-            ->setAccessibleTo('');
-
-        // Generate Contacts.
-        $contacts = new Contacts();
-
-        $contact = new ContactPersonIn();
-        $contact->setUserId('1');
-        $contact->setRoleCode('AUTHOR');
-        $contacts->addContact($contact);
-
-        $contact = new ContactPersonIn();
-        $contact->setUserId('2');
-        $contact->setRoleCode('EDITOR');
-        $contacts->addContact($contact);
-
-        $linguisticRequestIn = new LinguisticRequestIn();
-        $linguisticRequestIn->setContacts($contacts)
-            ->setGeneralInfo($generalInfo);
-
-        $createRequests = new CreateRequests();
-        $createRequests->setLinguisticRequest($linguisticRequestIn);
+        $createRequests = RequestsSerializer::fromArray(
+            $input['request'],
+            CreateRequests::class
+        );
```

Comment:
See comment above, we should have the expectations bundled together with the "input" test data, then we need to run assertions on those.

Reply:
Fixed.